ckanext/ontario_theme/fanstatic/scripts/odc-clear-filters.js

```diff
@@ -1,21 +1,27 @@
-(function () {
-    var clearFiltersButton = document.getElementById("clear-filters-button");
-    var facetsSelected = document.querySelectorAll(".filtered.pill");
+// Enable JavaScript's strict mode. Strict mode catches some common
+// programming errors and throws exceptions, prevents some unsafe actions from
+// being taken, and disables some confusing and bad JavaScript features.
 
-    if (facetsSelected.length >= 2) {
-        clearFiltersButton.style.display = "inline-block";
-        clearFiltersButton.addEventListener('click', clearFilters);
-    }
-    function clearFilters() {
-        const url = window.location.href.split('?')[0];
-        let params = new URLSearchParams(document.location.search);
-        let array = Array.from(params);
-        for (const field of array) {
-            const [key, value] = field;
-            if ((key != 'q')) {
-                params.delete(key);
-            }
-        }
-        window.location.href = `${url}?${params}`;
+/*
+ * CKAN module function for submitting form with empty facets once the clear
+ * filters button is clicked
+ */
+
+"use strict";
+
+ckan.module('clear_filters', function ($) {
+  return {
+      initialize: function () {
+          const facetsSelected = $(".filtered.pill");
```

Comment:
The code inside `initialize` is not necessary since the button will not be created in `search_form.html` (line 130) unless 2 or more facets have been selected.